pkg/webhook/immutable_fields_validator.go

```diff
@@ -380,6 +382,21 @@ func validateImmutableFieldsForLoggingLogMetricResource(oldSpec, spec map[string
 	return allowedResponse
 }
 
+func validateImmutableFieldsForSQLInstanceResource(oldSpec, spec map[string]interface{}) admission.Response {
+	ImmutableFields := []string{"cloneSource"}
```

Comment:
It's out of scope, though we probably need to think about how to handle use cases when a user realizes they made a mistake and want to fix it in-place after the creation/clone failed. Making the field strictly immutable means a user needs to delete the object, edit the manifest, and reapply, and could be a bit troublesome for some GitOps workflow.

Reply:
I think that we should require delete + recreate, rather than build in some special other mechanism, but maybe we can wait for product feedback on this.

Reply:
I think it's more idiomatic to require create + recreate. But maybe this could be partially solved by educating customer about k8s best practices. Or, perhaps there is some k8s-native pattern here the customer had suggested that I am unaware of.

Reply:
fixed